source/actors/Walk.gd

```diff
@@ -0,0 +1,10 @@
+extends "res://actors/PlatformPhysics.gd"
+
+export (String) var action_left = "walk_left"
+export (String) var action_right = "walk_right"
+export (float) var speed = 400.0
+
+func _physics_process(delta):
+	var movement_direction = (-int(Input.is_action_pressed(action_left)) 
```

Comment:
Take rid of Input checking in these classes, they are supposed to be reused in AI as well

Reply:
i would use a virtual function like 
`func control(delta):
   pass`
and make use of it in the `_physics_process`
So when you extends the class, the only thing you should do is to overwrite the `control` function

Reply:
Thanks! I updated the class already, by removing the input handling it seems to be more reliable! Check it out

https://github.com/pigdevstudio/kitchen-tales/pull/8/files#diff-f77a6670018bb28bdee4e0310a3cb5d3